gulp/tasks/jekyll-build-preview.js

```diff
@@ -0,0 +1,29 @@
+(function() {
+    'use strict';
+
+    var gulp = require('gulp'),
+        gitUtils = require('../util/gitUtils'),
+        childProcess = require('child_process'),
+        previewConfigFile = '_tmp_preview_config.yml',
+        previewSiteDir = '_preview_site';
+
+    gulp.task('jekyll-build-preview', function() {
+        var branch = gitUtils.currentBranch(),
+            previewBaseUrl = '/' + branch + '/';
+        // Create a temporary Jekyll configuration file which specifies the base URL for the preview site
+        childProcess.execSync(
```

Comment:
@dan-f Unfortunately the ``--baseurl`` parameter is to the Jekyll serve command but not to build (that's what I tried first). That's why I had to go with this somewhat ugly hack of a temporary file.

It's an interesting point about adding error handling to this function. My thinking is that if there is an error then the overall gulp task will fail with an error message. I think adding error handling to every command will make the code much harder to understand.

I'm starting to think that Gulp is a bit heavyweight and that this would have all been easier with a makefile. You don't typically have to write error handling in makefiles. You just chain together Unix commands. You also don't have to mess around with Gulp plugins for every task you want to do.